writer.cpp

```diff
@@ -365,7 +378,8 @@ void Writer::write_byte(unsigned int byte) {
     }
 }
 
-void Writer::record_byte(unsigned int byte) {
+void Writer::record_byte(unsigned int byte, unsigned long cc) {
+    if(cc) {}
```

Comment:
It looks like this method should be reverted.

Reply:
> It looks like this method should be reverted.


it fact it depends if we'd like to append the bytes and he cycle counter to the music_stream.
If we do this (which involves quite a bit of changes), we'll be able to use optimizations on the music_stream.

If you think there is no use case for it, I can revert.

Or just tell me if you don't intend to merge this PR at all.

Thx!

Reply:
Oh! I forgot about the option to dump optimized output. Is it something you need/want?

As a feature, it is maybe less compelling, the code gets more complicated and the cycle counter will only be an approximation.